### PR TITLE
Rewrite/reformat of "checks.md"

### PR DESCRIPTION
**Pull request by johndmulhausen**
The DO product docs team gave this file some love and are now publishing it as part of the product docs themselves. This PR contains the result.

See: 
https://www.digitalocean.com/docs/kubernetes/resources/clusterlint-errors/

### CODE REVIEW

File: checks.md [hunk 1]
```diff
@@ -510,24 +485,3 @@ spec:
       - name: nginx
         image: nginx:1.7.9
 ```
-
-###### Node Labels and Taints
```
Comment (adamwg) [inline checks.md]: This is a recently-added check - looks like it got deleted in a merge/rebase.

File: checks.md [hunk 2]
```diff
@@ -1,15 +1,16 @@
-###### Default Namespace
+[Clusterlint](https://github.com/digitalocean/clusterlint) flags issues with a cluster that might prevent or complicate its upgrade to a new Kubernetes version.
 
-Name:         `default-namespace`
+## Default Namespace
 
-Group:        `basic`
+- Name: `default-namespace`
+- Group: `basic`
```
[the remark this answers is not in the document]
Comment (johndmulhausen) [reply checks.md]: Done

File: checks.md (hunk 1 above)
Comment (johndmulhausen) [reply checks.md]: Re-added and given some love

File: checks.md [hunk 3]
```diff
@@ -137,46 +134,44 @@ spec:
...
 
-Description: Docker is the most popular runtime for Kubernetes. However, Kubernetes supports other container runtimes as well: containerd, CRI-O, etc. If the registry is not prepended to the image name, docker assumes `docker.io` and pulls it from DockerHub. However, the other runtimes will result in errors while pulling images. In order to maintain portability, we recommend to provide a fully qualified image name. If the underlying runtime is changed and the object configs are deployed to a new cluster, having fully qualified image names ensures that the applications don't break.
+Docker is the most popular runtime for Kubernetes. However, Kubernetes supports other container runtimes as well, such as containerd and CRI-O. If the registry is not prepended to the image name, docker assumes `docker.io` and pulls it from DockerHub. However, the other runtimes will result in errors while pulling images. To maintain portability, we recommend using a fully qualified image name. If the underlying runtime is changed and the object configs are deployed to a new cluster, having fully qualified image names ensures that the applications don't break.
```
[the remark this answers is not in the document]
Comment (johndmulhausen) [reply checks.md]: Fixed

**Comment by johndmulhausen** [comment]
> Thanks for doing this! The formatting and wording is much improved.
> 
> I've left a bunch of suggestions - most of them are "while you're here" fixes for pre-existing issues. Feel free to reword my suggestions, of course, and punt on anything you'd rather handle (or have someone else handle) as a follow-up.

You are most welcome - I think I've handled everything you pointed out. Let me know if anything else needs work!

**Pull request merged**